### ISSUE TITLE
pushing to oras (harbor) with 3.8 causes freeze/hang on fallback workaround

### ISSUE THREAD

**Issue by dcsouthwick** [apptainer/singularity#6051]
### Version of Singularity:

`3.8.0`

### Expected behavior
Push to oras _harbor_ endpoint using fallback that worked in 3.8pre*

Or at least fail, and print timeout or something reasonable

### Actual behavior

push goes for the retry workaround , then just hangs. ctrl+C has no effect, the terminal is locked.

### Steps to reproduce this behavior
Install 3.8, have a HARBOR instance with OIDC
rm -rf $HOME/.singularity
```
singularity login -u <user> -p <harbor CLI secret> oras://your.harbor.instance

singularity -d push $HOME/alpine_latest.sif oras://your.harbor.instance/test/alpine:latest
DEBUG   [U=1000,P=18264]   persistentPreRun()            Singularity version: 3.8.0
DEBUG   [U=1000,P=18264]   persistentPreRun()            Parsing configuration file /usr/local/etc/singularity/singularity.conf
DEBUG   [U=1000,P=18264]   handleConfDir()               /home/dsouthwi/.singularity already exists. Not creating.
DEBUG   [U=1000,P=18264]   handleRemoteConf()            Ensuring file permission of 0600 on /home/dsouthwi/.singularity/remote.yaml
DEBUG   [U=1000,P=18264]   Init()                        Image format detection
DEBUG   [U=1000,P=18264]   Init()                        Check for sandbox image format
DEBUG   [U=1000,P=18264]   Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=1000,P=18264]   Init()                        Check for sif image format
DEBUG   [U=1000,P=18264]   Init()                        sif image format detected
DEBUG   [U=1000,P=18264]   UploadImage()                 ORAS push not accepted, retrying without config for registry compatibility

<wait 1-1000 minutes>
 
^CDEBUG   [U=1000,P=18264]   func2()                       User requested cancellation with interrupt
^C^C^C^C^C^C^C^C^C^C^C^C
<program has froze/locked & unresponsive to interrupts>
```
### What OS/distro are you running

Ubuntu 20.*
Centos 7/8
Fedora 30+

### How did you install Singularity

EPEL, fedora, git build, binaries

**Comment by DrDaveD** [comment]
This is related to #5691 and recently merged #5951.  It was dctrudg that worked on it, and he has since left the project to instead work on a [https://github.com/sylabs/singularity](SingularityCE) fork of this project.  I suggest trying it with their 3.8.0; does it work there and not in the hpcng fork?  If so, then hopefully somebody still here can figure out what the difference is.  If it doesn't work there either, then I suggest posting the issue there and if they come up with a fix we will port it here.

**Comment by dcsouthwick** [comment]
Thanks, I'll followup there

**Comment by DrDaveD** [comment]
This has now been fixed in singularity-ce 3.8.4 release, discussed in sylabs/singularity#113.   The question now is whether there should also be a hpcng/singularity 3.8.4 or if it should wait for 3.9.0.

**Comment by kmuriki** [comment]
Thanks for your contributions. I will go ahead and close this issue now.
